# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../wxsvg/wxsvg-1.0.8-1-ffmpeg-avutil.h.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fix to check for avutil.h in a path where newer versions of ffmpeg install the header
Origin: OpenSDE (http://opensde.org)

This fix also implies the to explicitely pass -D__STDC_CONSTANT_MACROS to get
the UINT64_C macro from stdint.h which is used in libavutil/common.h.

Furthermore the check for avutil.h at the old location has been preserver, but
is only used as a fallback when there was libavutil/avutil.h was found.

--- wxsvg-1.0.8-1/configure.in.orig	2011-03-10 21:22:55.000000000 +0100
+++ wxsvg-1.0.8-1/configure.in	2011-04-17 16:17:07.364077975 +0200
@@ -176,8 +176,21 @@
 AM_CONDITIONAL(USE_FFMPEG, [test x$enable_ffmpeg = xyes])
 if test x$enable_ffmpeg = xyes; then
   PKG_CHECK_MODULES(FFMPEG, [libavformat >= 50.0.0 libavcodec >= 51.0.0 libavutil >= 49.0.0 libswscale])
-  # check path for old ffmpeg
-  AC_CHECK_HEADERS([ffmpeg/avutil.h]) 
+  # check path for new ffmpeg
+  # we need to explicitely pass -D__STDC_CONSTANT_MACROS to get the UINT64_C macro from stdint.h which
+  # is used in libavutil/common.h
+  CXXFLAGS="$CXXFLAGS -D__STDC_CONSTANT_MACROS"
+  AC_CHECK_HEADERS([libavutil/avutil.h],
+  [],
+  [
+    # check path for old ffmpeg
+    AC_CHECK_HEADERS([ffmpeg/avutil.h])
+  ],
+  [[
+  #ifdef HAVE_STDINT_H
+  # include <stdint.h>
+  #endif
+  ]])
   CXXFLAGS="$CXXFLAGS -DUSE_FFMPEG $FFMPEG_CFLAGS"
   LIBS="$LIBS $FFMPEG_LIBS"
 fi
